sky/clouds/azure.py

```diff
@@ -198,6 +198,16 @@ def get_image_size(cls, image_id: str, region: Optional[str]) -> float:
             # not the image location. Marketplace image is globally available.
             region = 'eastus'
         publisher, offer, sku, version = image_id.split(':')
+        # Since the Azure SDK requires explicitly specifying the image version number,
+        # when the version is "latest," we need to manually query the current latest version.
+        if version == 'latest':
+            versions = compute_client.virtual_machine_images.list(
+                location=region,
+                publisher_name=publisher,
+                offer=offer,
+                skus=sku)
+            latest_version = max(versions, key=lambda x: x.name)
```

Comment:
Did you confirm that the `name` field here is the image version number? I assume it is a string, so will `max()` work correctly? E.g. we need to make sure that `1.0.10` > `1.0.2`.

Reply:
Thanks for pointing out the `max()` problem, it should use `package.version` to parse the image version here. 
Also I tried to find out the meaning of the `name` int he [azure document](https://learn.microsoft.com/en-us/python/api/azure-mgmt-compute/azure.mgmt.compute.v2021_04_01.models.virtualmachineimageresource?view=azure-python), but it's not clear enough. 
So when I tested my pr, I debug the returned `versions` during the runtime, and the returned `name` here was acutuall the image version number.